### ISSUE TITLE
Mainland & al. 2015 Incorrect molecules.csv

### ISSUE THREAD

**Issue by Maxence-Lalis** [pyrfume/pyrfume-data#76]
Hi,

Great project ! It is an awesome work done here. I just maybe noted a small issue on Mainland & al. 2015 that i wanted to report : The final molecules.csv has incorrect Odor Code and Molecule association. 

It should come from the merging done on the main. Merging is done on index, and should be done on CID. Moreover, CIDs are re-computed even though they are already present in raw molecule file (Odor.tsv), and sometimes lead to wrongful new CID.

Finally, the merge implicitly delete mixtures. Maybe it's intentional. 
Also, just for information, in the file 'Odor.tsv', there are several times the same molecules for different odor codes.

I suggest we can fix the merging problem by replacing in main:
```
odorants = pd.read_csv('Odors.tsv', sep='\t')

molecules = pd.DataFrame(from_cids(odorants[~odorants.CID.isna()]['CID'].astype(int)))

molecules = molecules.merge(odorants, on='CID') #Outer merge if mixtures are wanted

molecules = molecules.drop(['SMILES'], axis=1).drop_duplicates(subset=['Odor'])
molecules = molecules.rename(columns={'Odor': 'Odorant'})
molecules.to_csv('molecules.csv')
```

Cheers,
Maxence Lalis

**Comment by rgerkin** [comment]
@Maxence-Lalis 
Thanks for the correction!  I am planning an audit before a release announcement later this year to catch more issues like this.  Would you like to submit a PR to make this change or would you like me to make it?

**Comment by Maxence-Lalis** [comment]
Thanks for your response! I would more than happy to submit a PR and make this change.

Moreover, I wanted to add that while testing this get_cids function (https://github.com/pyrfume/pyrfume/blob/main/pyrfume/odorants.py#L411), I noticed that sometimes instead of the CID of the isomer we get the CID of the racemic mixture. It could be seen by name (or another identifier). I understand it's a problem in data, but we should be aware of it. Maybe we can add an optional identifier in order to cross-check CID ?

**Comment by rgerkin** [comment]
I look forward to the PR.  

`get_cids` is fundamentally limited by the accuracy of the PubChem REST API in its response to the query, and how it orders the hits, so that is not something we have much control over.  If you did want to check the quality of the result, I could imagine the logic might be very complex; for example you could check if the input and output SMILES matched, but there are a number of reasons they might not match even if the CID was correct (e.g. one used rdkit and one used some other program).  The problem would even worse for names.  But if there is a specific, consistent error that PubChem REST API makes then that is something we could temporarily patch in Pyrfume (or in pubchempy) and then ask PubChem to fix long term.

I'm actually building a new exchange identifier service to be more robust to this.

**Comment by Maxence-Lalis** [comment]
Once again, thanks for your reply. Indeed to check, the logic can be very complex and may not even be correct at the end.
Looking forward your robust new exchange identifier, I'm sure it would save a lot of people from spending hours of manually checking the molecules!

**Comment by rgerkin** [comment]
Fixed in #77.
